orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -270,10 +277,12 @@ def forward(self, *inputs, **kwargs):
         new_input_shape = [list(input.size()) for input in inputs if input is not None]
         if self._current_input_shape is None or self._current_input_shape != new_input_shape:
             self._current_input_shape = new_input_shape
-            self._split_training_graph(*inputs, **kwargs)
-        elif self._device_changed:
+            self._build_training_graph()
             self._create_training_session()
-            self._device_changed = False
+        # TODO: disabled for now, since it caused a bug in NVBert fp32 run
+        # elif self._device_changed:
+        #     self._create_training_session()
+        #     self._device_changed = False
```

Comment:
We need to determine why the backend doesn't support multiple calls to `_create_training_session`

Reply:
When creating a new InferenceSession, there is a bug for destructing the original InferenceSession. I also added this in the comment.